site/content/blog/ytt-validations-preview.md

```diff
@@ -0,0 +1,124 @@
+image: /img/ytt.svg
+tags: ['ytt', 'data values', 'validation', 'preview', 'experiments']
+---
+
+### 📣 Announcing! 📣
+We are excited to announce that in `ytt` v0.41.0, we are including a _preview_ of a powerful new feature:
+
+**`ytt` Validations!**
+
+### What are `ytt` Validations?
+
+Validations are constraints that you can declare on your Data Values to ensure that what's supplied is not just well-formed, but in the expected range of values.
+
+Like this:
+
+```yaml
+#@data/values-schema
+---
+volumeSnapshotLocation:
+  spec:
+    #@schema/validation one_of=["aws", "azure", "vsphere"]
+    provider: ""
+```
+
+That one new annotation — `@schema/validation` — will:
+- validate the final data value against that rule (`provider:` set to one of the three values),
+- when exporting schema, translate the rule to OpenAPI v3 / JSON Schema validation.
+- automatically include documentation describing the rule.
+
+While we plan on delivering a host of useful named rules, out of the box, there is a way to define custom validation rules, yourself.
+
+For all the juicy details, check out the [Validations Proposal/Specification](https://github.com/vmware-tanzu/carvel/blob/004-schema-validation/proposals/ytt/004-schema-validation/README.md).
+
+This kind of thing is very useful for situations where a mis-configured deployment could waste a lot of resources and be much more difficult to troubleshoot than getting an immediate error message that the configuration value was wrong.
+
+### 🔬 It's an Experiment! 🧪
+
+To make a preview possible, we're also introducing a feature for enabling "**experiments**" in v0.41.0.
+
+An "experiment" is a feature that is still in development. By default, all experiments are off. 
+
+When you switch on an experiment, it lights-up the named feature, allowing you to take it for a spin. Of course, like any early access software, using `ytt` with any experiments enabled means it's not suitable for production environments. All APIs introduced with the experiment are unstable and will change as we improve the UX and behavior of the feature.
+
+So, `ytt` Validations is the first of these such experiments.
+
+We're doing this so that you have a voice in making sure features like this meet _your_ needs.
+
+### How do I get started?
+
+`ytt` v0.41.0 will be released soon. When it does, grab yourself the latest.
+
+Then, to enable Validations, and start including validation rules in a copy of _your_ schema:
+
+```yaml
+#@data/values-schema
+---
+dex:
+  #@schema/validation ("non-empty value", lambda v: len(v) > 1)
+  username: ""
+  #@schema/validation ("non-empty value", lambda v: len(v) > 1), ("not 'default'", lambda v: v != "default")
+  namespace: ""
+  #@schema/validation ("one of: aws, azure, vsphere", lambda v: v in ["aws", "azure", "vsphere"])
+  provider: ""
+```
+
+... and set the OS environment variable, `YTTEXPERIMENTS`...
+
+```console
+$ YTTEXPERIMENTS=validations yttt -f schema.yml -v dex.namespace=default
```

Comment:
extra t in `ytt`. Happens in a couple other places as well.

Reply:
oops! Yeah, I have an alias `yttt` that executes my locally built version. 😬 

Will update.